frontend/new-admin-vm/Install-CAM.ps1

```diff
@@ -976,6 +1037,69 @@ graphURL=https\://graph.windows.net/
 				$laSecretVersionedURL = $laSecret.Id
 				$laSecretURL = $laSecretVersionedURL.Substring(0, $laSecretVersionedURL.lastIndexOf('/'))
 
+				################################
+				Write-Host "Populating user blob"
+				################################
+				$container_name = "cloudaccessmanager"
+				$acct_name = $using:userDataStorageAccount
+				$new_agent_vm_files = @(
+					"Install-PCoIPAgent.ps1", 
+					"Install-PCoIPAgent.sh", 
+					"$using:linuxAgentARM", 
+					"$using:gaAgentARM",
+					"$using:agentARM", 
+					"$using:sumoAgentApplicationVM",
+					"$using:sumoConf",
+					"InstallPCoIPAgent.ps1.zip",
+					"$using:idleShutdownLinux"
+					)
+				Write-Host "Will upload these files: $new_agent_vm_files"
+				$acctKey = (Get-AzureRmStorageAccountKey -ResourceGroupName $using:RGName -AccountName $acct_name).Value[0]
+				$ctx = New-AzureStorageContext -StorageAccountName $acct_name -StorageAccountKey $acctKey
+				try {
+						Get-AzureStorageContainer -Name $container_name -Context $ctx -ErrorAction Stop
+				} Catch {
+						# -Permission needs to be off to allow only owner read and to require access key!
+						New-AzureStorageContainer -Name $container_name -Context $ctx -Permission "Off"
+				}
+				Write-Host "Uploading files to private blob"
+				ForEach($file in $new_agent_vm_files) {
+						$filepath = Join-Path $using:LocalDLPath $file
+						try {
+							Get-AzureStorageBlob -Context $ctx -Container $container_name -Blob "remote-workstation/$file" -ErrorAction Stop
+						# file already exists do nothing
+						} Catch {
+							Write-Host "Uploading $filepath to blob.."
+							Set-AzureStorageBlobContent -File $filepath -Container $container_name -Blob "remote-workstation\$file" -Context $ctx
+						}
+				}
+
+				$blobUri = (((Get-AzureStorageBlob -Context $ctx -Container $container_name)[0].ICloudBlob.uri.AbsoluteUri) -split '/')[0..4] -join '/'
+
+				# this is the url to access the blob account
+				$blobUriSecretName = "userStorageAccountUri"
+				Set-AzureKeyVaultSecret -VaultName $kvName -Name $blobUriSecretName -SecretValue (ConvertTo-SecureString $blobUri -AsPlainText -Force) -ErrorAction stop
+
+				$storageAccountSecretName = "userStorageName"
```

Comment:
Do we want to use the word "user" for this? "user" sounds like end-user, but this isn't really end-user storage. maybe CAMStorageName?

Reply:
if you think about it though all the templates stored there are for the end-user. The user customizes them and stores them there, so i named it userStorage :)